src/nvme/ioctl.c

```diff
@@ -355,6 +355,130 @@ int nvme_get_log_page(int fd, __u32 xfer_len, struct nvme_get_log_args *args)
 	return 0;
 }
 
+static int read_ana_chunk(int fd, enum nvme_log_ana_lsp lsp, bool rae,
+			  __u8 *log, __u8 **read, __u8 *to_read, __u8 *log_end)
+{
+	if (to_read > log_end) {
+		errno = ENOSPC;
+		return -1;
```

Comment:
Internal function can just return the error code. We will drop the POSIX style of error codes in the next mayor library version.

Reply:
We do this in the rest of the code (or should), that is only the function which are exported should touch errno. The reason I want to keep it this is way, is that when we drop the errno usage, we don't have to review internal functions.

Reply:
Makes sense. Let's fix this after this merge.